react/features/security/components/security-dialog/native/SecurityDialog.js

```diff
@@ -0,0 +1,408 @@
+import { translate } from '../../../../base/i18n';
+import {
+    isLocalParticipantModerator
+} from '../../../../base/participants';
+import { StyleType } from '../../../../base/styles';
+import { toggleLobbyMode } from '../../../../lobby/actions.any';
+import LobbyModeSwitch
+    from '../../../../lobby/components/native/LobbyModeSwitch';
+import { LOCKED_LOCALLY, LOCKED_REMOTELY } from '../../../../room-lock';
+import {
+    endRoomLockRequest,
+    unlockRoom
+} from '../../../../room-lock/actions';
+import RoomLockSwitch from '../../../../room-lock/components/RoomLockSwitch';
+
+/**
+ * The style of the {@link TextInput} rendered by {@code SecurityDialog}. As it
+ * requests the entry of a password, {@code TextInput} automatically correcting
+ * the entry of the password is a pain to deal with as a user.
+ */
+const _TEXT_INPUT_PROPS = {
+    autoCapitalize: 'none',
+    autoCorrect: false
+};
+
+/**
+ * The type of the React {@code Component} props of {@link SecurityDialog}.
+ */
+type Props = {
+
+    /**
+     * The JitsiConference which requires a password.
+     */
+    _conference: Object,
+
+    /**
+     * The color-schemed stylesheet of the feature.
+     */
+    _dialogStyles: StyleType,
+
+    /**
+     * State of the lobby mode.
+     */
+    _lobbyEnabled: boolean,
+
+    /**
+     * Whether the lobby mode switch is available or not.
+     */
+    _lobbyModeSwitchVisible: boolean,
+
+    /**
+     * The value for how the conference is locked (or undefined if not locked)
+     * as defined by room-lock constants.
+     */
+    _locked: string,
+
+    /**
+     * The current known password for the JitsiConference.
+     */
+    _password: string,
+
+    /**
+     * Number of digits used in the room-lock password.
+     */
+    _passwordNumberOfDigits: number,
+
+    /**
+     * Whether the room lock switch is available or not.
+     */
+    _roomLockSwitchVisible: boolean,
+
+    /**
+     * The color-schemed stylesheet of the security dialog feature.
+     */
+    _securityDialogStyles: StyleType,
+
+    /**
+     * Redux store dispatch function.
+     */
+    dispatch: Dispatch<any>,
+
+    /**
+     * Invoked to obtain translated strings.
+     */
+    t: Function
+};
+
+/**
+ * The type of the React {@code Component} state of {@link SecurityDialog}.
+ */
+type State = {
+
+    /**
+     * Password added by the participant for room lock.
+     */
+    passwordInputValue: string,
+
+    /**
+     * Shows an input or a message.
+     */
+    showElement: boolean
+};
+
+/**
+ * Component that renders the security options dialog.
+ *
+ * @returns {React$Element<any>}
+ */
+class SecurityDialog extends Component<Props, State> {
+
+    /**
+     * Instantiates a new {@code SecurityDialog}.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            passwordInputValue: '',
+            showElement: Boolean(
```

Comment:
I called it showElement because we also show the message regarding the password set by another participant and I thought element was something more general.

Reply:
Ah, got it.